<?xml version="1.0" encoding="utf-8"?>
<!-- Launch file for online Relative Deviation analysis -->

<launch>

  <arg name="cad_file" default="$(find cpt_selective_icp)/resources/cla.json" />
  <arg name="map_frame" default="/map" />
  <arg name="discrete_color" default="false" />
  <arg name="score_threshold" default="0.0015" />
  <arg name="path" default="$(find cpt_deviation_analysis)" />
  <arg name="cad_topic" default="mesh_publisher/mesh_out" />
  <arg name="scan_topic" default="ref_corrected_scan" />
  <arg name="map_topic" default="map" />

  <node name="relative_dev"
        pkg="cpt_deviation_analysis"
        type="relative_deviations_node"
        output="screen"
        respawn="false">
    <rosparam command="load" file="$(find cpt_deviation_analysis)/launch/parameters.yaml" />
    <param name="map_frame" value="$(arg map_frame)" />
    <param name="discrete_color" value="$(arg discrete_color)" />
    <param name="score_threshold" value="$(arg score_threshold)" />
    <param name="path" value="$(arg path)" />
    <param name="cad_topic" value="$(arg cad_topic)" />
    <param name="scan_topic" value="$(arg scan_topic)" />
    <param name="map_topic" value="$(arg map_topic)" />
  </node>

  <node name="mesh_positioning"
        pkg="cpt_selective_icp"
        type="position_cad_with_marker.py"
        output="screen">
    <param name="marker_parent_frame" value="map" />
  </node>

	<node name="find_id_in_mesh"
        pkg="cpt_selective_icp"
        type="find_closest_id_in_mesh.py"
        output="screen">
  </node>

  <node name="mesh_publisher"
        pkg="cpt_deviation_analysis"
        type="mesh_publisher_node"
        output="screen">
    <param name="publish_on_start" value="true"/>
    <param name="default_filename" value="$(arg cad_file)"/>
    <param name="frame_name" value="marker_position"/>
  </node>
</launch>
